app/assets/javascripts/shopify_app/itp_polyfill.js

```diff
@@ -0,0 +1,27 @@
+function shouldTriggerCookiePartitioning() {
+  var userAgent =  window.navigator.userAgent;
+  var isSafari = userAgent.indexOf('Safari');
+  var versionNumber = isSafari ? parseFloat(userAgent.match(/Version\/(\d+\.?\d*)/)[1]) : null;
+  
+  // TODO: Replace with library for checking user-agents
+  return document.hasStorageAccess && isSafari && versionNumber >= 12;
+}
+
+function setCookieAndRedirect() {
+  document.cookie = "shopify.cookies_persist=true";
+  window.location.href = window.shopOrigin + "/admin/apps/" + window.apiKey;
+
+  return false;
```

Comment:
You don't need this. The button you've attached the "click" event at, it's `type="button"`, therefore won't do any for submission or navigation. `return false` to prevent clicks is generally considered to be a bad practice (due to stopping event propagation)